### ISSUE TITLE
[GR-45886] Register L32X64MixRandom constructor for reflection as needed

### ISSUE THREAD

**Issue by galderz** [oracle/graal#6456]
`jdk.random.L32X64MixRandom` constructor should be registered for reflection by GraalVM. This is a new JDK class introduced in Java 17. Users expect reflection for such JDK classes to be done by GraalVM itself.

I'm assuming that the class has not yet been registered when needed because it was added in a relatively new JDK.

As a workaround, the user can register the constructor manually via the reflection json configuration.

This could be a good first issue (I can't add labels).

**Comment by zakkak** [comment]
> `jdk.random.L32X64MixRandom` constructor should be registered for reflection by GraalVM.

I think the class should not be unconditionally registered as most Java applications are not expected to require it.

Instead it should be conditionally registered when `java.util.random.RandomGeneratorFactory#getDefault()` or `java.util.random.RandomGenerator#getDefault()` is reachable.

See https://github.com/openjdk/jdk17u/blob/2fe42855c48c49b515b97312ce64a5a8ef3af407/src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java#L371 and https://github.com/openjdk/jdk17u/blob/2fe42855c48c49b515b97312ce64a5a8ef3af407/src/java.base/share/classes/java/util/random/RandomGenerator.java#L138

**Comment by oubidar-Abderrahim** [comment]
Thank you for reporting this, The issue is tracked internally on GR 45886

**Comment by zakkak** [comment]
Hello @oubidar-Abderrahim , is there any progress on that?

**Comment by loicottet** [comment]
`RandomGeneratorFactory` has been rewritten to not use reflection, I was not able to reproduce this issue anymore.

**Comment by charpov** [comment]
> `RandomGeneratorFactory` has been rewritten to not use reflection, I was not able to reproduce this issue anymore.

Which Java is that? I tried with Java 21 a few week ago and the problem was still there. Are you talking about Java 25?

**Comment by loicottet** [comment]
Yes, Java 25 doesn't access `jdk.random.L32X64MixRandom` through reflection as far as I can see.
